datasetinsights/scripts/public_download.py

```diff
@@ -0,0 +1,89 @@
+""" Download Public GroceriesReal Dataset from GCS
+"""
+import argparse
+import logging
+import os
+
+from datasetinsights.data.datasets import GroceriesReal
+from datasetinsights.data.simulation.download import (
+    download_file_from_url,
+    compare_checksums,
+)
+import datasetinsights.constants as const
+
+LOCAL_PATH = "groceries"
+
+logging.basicConfig(
+    level=logging.INFO,
+    format=(
+        "%(levelname)s | %(asctime)s | %(name)s | %(threadName)s | "
+        "%(message)s"
+    ),
+    datefmt="%Y-%m-%d %H:%M:%S",
+)
+logger = logging.getLogger(__name__)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="Download GroceriesReal dataset from a public GCS path"
+    )
+    parser.add_argument(
+        "-v",
+        "--verbose",
+        action="store_true",
+        default=False,
+        help="turn on verbose mode to enable debug logging",
+    )
+    parser.add_argument(
+        "--data-root",
+        type=str,
+        default=const.DEFAULT_DATA_ROOT,
+        help="root directory of datasets",
+    )
+    parser.add_argument(
+        "--name",
+        type=str,
+        default=const.DEFAULT_DATA_ROOT,
+        help="dataset name, e.g. GroceriesReal, Synthetic",
+        required=True,
+    )
+    parser.add_argument(
+        "--version", type=str, default="v3", help="dataset version, e.g. v3",
+    )
+
+    args = parser.parse_args()
+
+    return args
+
+
+def run(args):
```

Comment:
Can you add tests for this method?

Reply:
Since all the methods in `run()` have been tested and all the parameters are constant, I don't think it needs a unittest.

Reply:
We should mock up code that was already tested. However, there are still values to test how each of these methods are wired up together.

Reply:
Done.